hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -302,14 +303,19 @@ protected OmMetadataManagerImpl() {
   }
 
   // metadata constructor for snapshots
-  private OmMetadataManagerImpl(OzoneConfiguration conf, String snapshotDirName)
-      throws IOException {
+  private OmMetadataManagerImpl(OzoneConfiguration conf, String snapshotDirName,
+      boolean isSnapshotInCache) throws IOException {
     lock = new OmReadOnlyLock();
     omEpoch = 0;
     String snapshotDir = OMStorage.getOmDbDir(conf) +
         OM_KEY_PREFIX + OM_SNAPSHOT_DIR;
-    setStore(loadDB(conf, new File(snapshotDir),
-        OM_DB_NAME + snapshotDirName, true));
+    File metaDir = new File(snapshotDir);
+    String dbName = OM_DB_NAME + snapshotDirName;
+    if (isSnapshotInCache) {
+      File checkpoint = Paths.get(metaDir.toPath().toString(), dbName).toFile();
+      RDBCheckpointManager.waitForCheckpointDirectoryExist(checkpoint);
```

Comment:
Yup this is only to avoid reads failing right after a createSnapshot Request, Since we don't have a cache like feature in snapshot to prevent subsequent reads till entry is actually flushed to DB, I think all we can do is wait till entry gets flushed and if entry is in cache, it is most likely it will get flushed .

Reply:
One more idea is to create the checkpoint in OmCreateSnapshotCreateRequest#validateAndUpdateCache inside the bucket lock, but the problem here is that if OM crashes/shuts down during the time the entry is in cache and not flushed to DB, a waste checkpoint will be created since it won't be accessed anymore, i.e a failed createSnapshotRequest will create checkpoints and adds up to the space.

Reply:
I asked whether Rocksdb createCheckpoint call  is synchronous in the rocksdb community group and this is what I got.
> 'checkpoint is always sync  after the CreateCheckpoint is completed a point in time copy of your database exists in the output directory'

That aside, the problem of unnecessary checkpoint creation when snapshot create fails is still an issue.